### PR TITLE
Tooltips available for SCM diff decorations

### PR DESCRIPTION
**Pull request by marckassay**
This pull request is intended to complete feature request #25849.  The image below illustrates this feature (with my cheapo Paint cursor):

25849_sm

There are two caveats.  The first, @joaomoreno recommended to have this new tooltip property in the `SourceControlResourceThemableDecorations` interface of the API.  I found that to be nested one level too deep, so instead I created the property in the `SourceControlResourceDecorations`.

The second, I'm unsure if I'm breaking any project protocol by calling the `localize()` the way it is in `git/src/model.ts`.  Any key value will do?

Let me know if anything else is needed!

### CODE REVIEW

File: extensions/git/src/model.ts [hunk 1]
```diff
@@ -118,6 +118,22 @@ export class Resource implements SourceControlResourceState {
 		}
 	}
 
+	private get tooltip(): string | undefined {
```
Comment (joaomoreno) [inline extensions/git/src/model.ts]: Although smart, this automatic labelling based on the name of the enum value won't work with localize. Please create a private `getTooltip` method, similar to `getIconPath`, where the mapping from Status to tooltip is done manually.

**Review by joaomoreno — CHANGES_REQUESTED** [review]

File: extensions/git/src/model.ts (hunk 1 above)
Comment (marckassay) [inline extensions/git/src/model.ts]: I see.  I had some build issues on my machine today, so I will attempt tomorrow.

**Comment by joaomoreno** [comment]
@marckassay You might want to start over. You have very likely ran `git rebase` after merging upstream. This causes many duplicate commits in the PR. I'll close this. Please recreate the PR with a clean history. 👍

**Comment by marckassay** [comment]
@joaomoreno that's exactly what I did; inadvertently rebased.

Ok, let me know if the revised PR (#29644) needs anything adjusted.

**Pull request closed without merging**